lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -158,21 +155,39 @@ class QcFlagService {
      * @return {Promise<void>} resolve once all periods are updated
      */
     async _updateEffectivePeriods({ from: newerPeriodFrom, to: newerPeriodTo }, periods) {
+        /**
+         * When run's timestamps are missing, effective periods have null timestamps, therefore
+         * comparison must be evaluated on extended set of values
+         * so  `from` == null is extended to -1, which is interpreted as run.startTime
+         * and `to`   == null is extended to MAX_DATE, which is interpreted as run.endTime
+         */
+        const MAX_DATE = new Date('100000-01-01').getTime();
+        const newerFromEval = newerPeriodFrom ?? -1;
+        const newerToEval = newerPeriodTo ?? MAX_DATE;
+
         for (const effectivePeriod of periods) {
             const { id: effectivePeriodId, from: effectiveFrom, to: effectiveTo } = effectivePeriod;
 
-            if (newerPeriodFrom <= effectiveFrom && effectiveTo <= newerPeriodTo) { // Old flag is fully covered by new one
+            const evalEffectiveFrom = effectiveFrom ?? -1;
+            const evalEffectiveTo = effectiveTo ?? MAX_DATE;
+
+            if (newerFromEval <= evalEffectiveFrom && evalEffectiveTo <= newerToEval) { // Old flag is fully covered by new one
                 await QcFlagEffectivePeriodRepository.removeOne({ where: { id: effectivePeriodId } });
-            } else if (effectiveFrom < newerPeriodFrom && newerPeriodTo < effectiveTo) { // New flag's period is included in the old one's period
+            } else if (evalEffectiveFrom < newerFromEval
+                && newerToEval < evalEffectiveTo) {
+                /*
+                 * New flag's period is included in the old one's period.
+                 * Possible only when both timestamps of newer period are not null
+                 */
                 await QcFlagEffectivePeriodRepository.update(effectivePeriod, { to: newerPeriodFrom });
                 await QcFlagEffectivePeriodRepository.insert({
                     flagId: effectivePeriod.flagId,
                     from: newerPeriodTo,
                     to: effectiveTo,
                 });
-            } else if (effectiveFrom < newerPeriodFrom) {
+            } else if (evalEffectiveFrom < newerFromEval) { // Possible only when `newerPeriodFrom` is not null
```

Comment:
I am not sure to understand why this comment?

Reply:
Just to remind that as null is evaluated to -1 here, this condition can be true iff newerPeriodFrom is not null